src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1757,14 +1836,36 @@ void uct_ud_ep_disconnect(uct_ep_h tl_ep)
      * to not invoke user callbacks for the endpoint which was destroyed */
     uct_ud_ep_purge(ep, UCS_ERR_CANCELED);
 
+    if (!(ep->flags & UCT_UD_EP_FLAG_CONNECT_TO_EP) &&
+        (ep->flags & UCT_UD_EP_FLAG_TX)) {
```

Comment:
uct_ud_ep_disconnect is called for API endpoint, so how could it be without TX capability?

Reply:
it is called from `uct_ud_ep_handle_peer_disconnect` though `uct_ep_destroy`